<!-- Provide tesse-gym with data pulled from tesse_ros_bridge -->
<launch>

    <!-- Set ports -->
    <arg name="metadata_port"              default="9007"/>
    <arg name="image_port"                 default="9008"/>
    <arg name="ns"                         default="tesse"/>
    <arg name="use_ground_truth"           default="true"/>
    <arg name="run_segmentation_on_demand" default="true"/>
    <arg name="model_type"                 default="TRTModel"/>
    <arg name="weights"                    default="$(find tesse_segmentation_ros)/cfg/goseek-v0.1.2/v3-unet-resnet18-epoch-12.onnx"/>
    <arg name="publish_segmentation"       default="true"/>
    <arg name="publish_depth"              default="true"/>
    <arg name="publish_mono_stereo"        default="true"/>
    <arg name="publish_metadata"           default="true"/>


    <node name="tesse_gym_image_bridge" pkg="tesse_gym_bridge" type="image_server.py" output="screen">

        <!-- Set parameters -->
        <param name="image_port"                 value="$(arg image_port)"/>
        <param name="use_ground_truth"           value="$(arg use_ground_truth)"/>
        <param name="run_segmentation_on_demand" value="$(arg run_segmentation_on_demand)"/>
        <param name="model_type"                 value="$(arg model_type)"/>
        <param name="weights"                    value="$(arg weights)"/>
        <param name="publish_segmentation"       value="$(arg publish_segmentation)"/>
        <param name="publish_depth"              value="$(arg publish_depth)"/>
        <param name="publish_mono_stereo"        value="$(arg publish_mono_stereo)"/>

        <!-- remap subscriptions -->
        <remap from="/left_cam/image_raw"           to="/$(arg ns)/left_cam/image_raw"/>
        <remap from="/right_cam/image_raw"          to="/$(arg ns)/right_cam/image_raw"/>
        <remap from="/segmentation/image_raw"       to="/$(arg ns)/segmentation/image_raw"/>
        <remap from="/depth/image_raw"              to="/$(arg ns)/depth/image_raw"/>
        <remap from="/metadata"                     to="/$(arg ns)/metadata"/>
        <remap from="/segmentation_noisy/image_raw" to="/semantic_segmentation/prediction_class_colored"/>
        <remap from="/depth_noisy/image_raw"        to="/$(arg ns)/depth/image_raw"/>
        <remap from="/metadata"                     to="/$(arg ns)/metadata"/>
    </node>

    <node name="tesse_gym_metadata_bridge" pkg="tesse_gym_bridge" type="metadata_server.py" output="screen">
        <param name="metadata_port"    value="$(arg metadata_port)"/>
        <param name="use_ground_truth" value="$(arg use_ground_truth)"/>
        <remap from="/metadata"        to="/tesse/metadata"/>
    </node>
</launch>

